Sources/ElasticSwift/Networking/SessionManager.swift

```diff
@@ -69,42 +69,85 @@ class SessionManager: NSObject, URLSessionDelegate {
+            var currRequest = URLRequest(url: url)
+            currRequest.httpMethod = method.rawValue
+            if let bd = body {
+                currRequest.httpBody = bd.data(using: .utf8)
+            }
+            self.request = currRequest
+            debugPrint("URLRequest with body created \(self.request.debugDescription)")
+        } else {
+            print("Impossible to create URLRequest")
+        }
         return self
     }
     
-    func createRequest(method: HTTPMethod, forPath pathComponent: String, witParams: Any, body: Data) -> Self {
-        var currRequest = URLRequest(url: self.url.appendingPathComponent(pathComponent))
-        currRequest.httpMethod = method.rawValue
-        currRequest.httpBody = body
-        self.request = currRequest
-        debugPrint("URLRequest with body created")
+    func createRequest(method: HTTPMethod, forPath pathComponent: String, witParams queryItems: [QueryParams:String]?, body: Data?) -> Self {
+        if let url = url(fromBaseURL: self.url, path: pathComponent, queryItems: queryItems) {
+            var currRequest = URLRequest(url: url)
+            currRequest.httpMethod = method.rawValue
+            if body != nil {
+                currRequest.httpBody = body
+            }
+            self.request = currRequest
+            debugPrint("URLRequest with body created")
+        } else {
+            print("Impossible to create URLRequest")
+        }
         return self
     }
     
+    private func url(fromBaseURL baseURL:URL, path:String, queryItems:[QueryParams:String]?) -> URL? {
+        let url = baseURL.appendingPathComponent(path)
+        var components = URLComponents(url: url, resolvingAgainstBaseURL: true)
+        
+        if let items = queryItems {
+            var query = [URLQueryItem]()
+            for itemKey in items.keys {
+                if let itemValue = items[itemKey] {
+                    let qi = URLQueryItem(name: itemKey.rawValue, value: itemValue)
+                    query.append(qi)
+                }
+            }
+            components?.queryItems = query
+        }
+        
+        let result = components?.url
+        debugPrint("URL Created: ",result ?? "NIL")
+        return result
+    }
+    
     func createDataTask(onCompletion callback: @escaping (_ response: ESResponse) -> Void) -> Self {
-        self.dataTask = self.session?.dataTask(with: self.request!) { data, response, error in
+        guard let request = self.request else {
+            debugPrint("Data Task NOT Created! Request == NIL")
+            return self
+        }
+        
+        self.dataTask = self.session?.dataTask(with: request) { data, response, error in
             let response = ESResponse(data: data, httpResponse: response, error: error)
             return callback(response)
         }
         
-        debugPrint("Data Task Created:", self.dataTask!)
```

Comment:
I don't understand why you force unwrap optionals in debugging ...
In my opinion force unwrap is to avoid unless there's no other method to get it done.
We need to avoid the App crash but manage all possible cases ...